game/src/core/level/elements/ILevel.java

```diff
@@ -268,23 +273,38 @@ default void changeTileElementType(final Tile tile, final LevelElement changeInt
    * specified type is empty, the method returns null.
    *
    * @param elementType Type of the tile to retrieve.
-   * @return A random tile of the specified type, or null if the list for that type is empty.
+   * @return A random tile of the specified type, or empty if the list for that type is empty.
    */
-  default Tile randomTile(final LevelElement elementType) {
+  default Optional<Tile> randomTile(final LevelElement elementType) {
```

Comment:
In dieser Methode passiert ziemlich viel Code-Duplizierung. Es wird nach dem Tile-Typ unterschieden, dann jedesmal gefragt, ob die Liste leer ist oder nicht, und dann ein Tile berechnet oder ein leeres Optional.

Das leere Optional ist für alle Tile-Typen identisch. Das würde ich definitiv auslagern und für alle Tile-Typen nur einmal gemeinsam machen. 

Ebenso könnte man das Berechnen des gefüllten Optionals einmal für alle machen - dazu müsste man statt `return switch....` quasi sowas machen `return Optional.ofNullable(switch ...)` und im switch der alte Code. (Das geht so nicht, aber eine Hilfsvariable sollte gehen.)

Man könnte auch überlegen, hier einfach mit try/catch zu arbeiten: im try erzeugt man ohne Abfrage der Listenlänge ein Optional. Wenn die Liste leer ist, kommt man in das catch und gibt dort das leere Optional zurück. Das ist hier äußerst verlockend, auch wenn ich kein Fan davon bin, Exceptionhandling für die Kontrollflusssteuerung zu missbrauchen.

Reply:
die neuen (java21) switch-expressions erlauben auch guards, also sowas wie 

```
switch(x) {
    case String s when s.length>0 -> foo();
```

weiss aber grad nicht, ob sich die guards dann auf die variable aus dem case-pattern beziehen müssen oder beliebige ausdrücke sein dürfen ...

und im default-fall kannst du dann das leere optional erzeugen.

Reply:
@AMatutat Hier ein Code-Vorschlag, den ich leider nicht direkt als Vorschlag eintüten kann: 

```java
    try {
      return Optional.of(
          switch (elementType) {
            case SKIP -> skipTiles().get(RANDOM.nextInt(skipTiles().size()));
            case FLOOR -> floorTiles().get(RANDOM.nextInt(floorTiles().size()));
            case WALL -> wallTiles().get(RANDOM.nextInt(wallTiles().size()));
            case HOLE -> holeTiles().get(RANDOM.nextInt(holeTiles().size()));
            case EXIT -> exitTiles().get(RANDOM.nextInt(exitTiles().size()));
            case DOOR -> doorTiles().get(RANDOM.nextInt(doorTiles().size()));
            case PIT -> pitTiles().get(RANDOM.nextInt(pitTiles().size()));
          });
    } catch (Exception e) {
      // iff tile list is empty
      return Optional.empty();
    }
```

Reply:
Irgendwie ist GH durcheinander. Hier geht es weiter: https://github.com/Dungeon-CampusMinden/Dungeon/pull/1624#pullrequestreview-2245903323 ....

Reply:
Ich finde Variante (3) am besten. Keine Duplizierung mehr, keine unnötigen Hilfsvariablen mehr, und kein unschönes "Missbrauchen" des Exception-Handlings für Kontrollflusssteuerung :)

Reply:
Variante 3 commited

Reply:
👌